### DISCUSSION TITLE
How to update Rye's version of uv?

### DISCUSSION THREAD

**Discussion by cp2boston** [astral-sh/rye#996] [Q&A]
Rye and uv are improving at a pretty rapid pace.  Is there a way to upgrade the Rye install of uv?  The current uv is 0.1.28 and the Rye uv is 0.1.26.

**Comment by philippemiron** [comment]
Looks like `uv` is updated with new version of `rye`. If you look at the 0.32.0 version release (https://github.com/astral-sh/rye/releases/tag/0.32.0), it says:

- Update uv to 0.1.26.

**Reply by cp2boston** [reply]
Ah, thanks for pointing that out!  I'm hoping that the latest uv will resolve the --extra-index-url issue and it's at 0.1.28.  I'll be patient 😁

**Reply by CatamountJack** [reply]
Same here, waiting on uv 0.1.29 so I can install with x86 python.  If there was a way to update independent of rye that would be great.

**Reply by bluss** [reply]
It can't be recommended but for experimental purposes one can install uv separately and patch it into Rye (copy it into the right place inside $RYE_HOME). It could of course break it all, but right now uv it happens that it doesn't. Uv is also a single binary which is easily installable. This can be used for testing bug fixes. One can also check out rye source, update uv and build rye.
